gateway/assets/script/bootstrap.js

```diff
@@ -31,11 +41,34 @@ app.controller("home", ['$scope', '$log', '$http', '$location', '$timeout', '$md
             labels: {}
         };
 
-        $scope.invocation.request = "";
+        $scope.isReady = function(selectedFunction) {
+            return (selectedFunction.ready != undefined && selectedFunction.ready == true);
+        }
 
-        setInterval(function() {
+        $scope.invocation.request = "";
+        $scope.isFunctionReady = "Querying";
+        $scope.fetchFunctionsDelay = 3500;
+        $scope.queryFunctionDelay = 2500;
+        
+        $scope.fetchFunctionsInterval = setInterval(function() {
             refreshData();
-        }, 1000);
+        }, $scope.fetchFunctionsDelay);
+
+        $scope.queryFunctionInterval = setInterval(function() {
+            if($scope.selectedFunction && $scope.selectedFunction.name) {
```

Comment:
I have to see the details but this statement seems strange...

`$scope.selectedFunction && $scope.selectedFunction.name`

It looks like `selectedFunction` has multiple types (sometimes it's an `object` , and sometimes it's `string` ?)

Reply:
my bad... I mislooked and though it was

```
$scope.selectedFunction === $scope.selectedFunction.name
```

ignore this comment please